src/rust/lib/graph-editor/src/component/node.rs

```diff
@@ -214,6 +220,11 @@ impl Node {
             selection_ref.set_target_position(0.0);
         });
 
+        // TODO this is sample functionality. Needs to be replaced with logic creating ports.
+        self.data.port_manager.set_parent(self.downgrade());
+        self.data.port_manager.create_input_port();
+        self.data.port_manager.create_output_port();
```

Comment:
Plus, what is the API to rotate ports, now? How can I do it? Can I have a reference to a port and write `port.set_rotation(...)` ? The API should be like that.

Reply:
Since the positioning and rotating of the ports isn't independent the idea was for the `PortManager` to handle that functionality internally.

I can refine the API to allow access to individual ports though and allow someting like `node.ports.input.add()`

Reply:
What you mean by `isn't independent`? I know they are placed on a circle, thus I was asking about function allowing ports to set their rotation (by providing the angle only).

Reply:
If ports are positioned around the node their position needs to be updated and their rotation needs to be updated to point in the right direction. But this is now handled inside of the `Ports` themselves.